### ISSUE TITLE
Make home page support rendering the component of other plugins

### ISSUE THREAD

**Issue by raintygao** [opensearch-project/OpenSearch-Dashboards#5586]
**Is your feature request related to a problem? Please describe.**

![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/82c5e82e-8f9a-41b2-af74-0fe4a5e490e6)
My team is considering adding a component from external plugin to the home page(in the position of the horizontal rule line in the screenshot). When interacting with the component (such as clicking the button of this component), the function in the plugin can be triggered.

**Describe the solution you'd like**

The component includes functions inside the plugin, so it should be passed  from the plugin to home. We need an approach to pass this react node type component from the plugin to home. If the plugin is not registered, home does not need to be changed.
We are currently thinking of three approaches, the first two seem more straightforward. Please feel free to leave your comments.

1. Set this plugin as the optionalPlugin of home plugin, so that home can get this passed component in `setup` hook.
2. Set home plugin as the optionalPlugin of this plugin, so that this plugin can pass this component in the function returned from home in `start` hook of this plugin.
3. If OSD core has a public service supports functions like `get` and `set`, plugin can call `set` to pass component and home can call `get to get component and render it.

**Comment by kavilla** [comment]
I agree! I created a sub-issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5589 for the new home plugin view.

That touches upon this. Where 3rd party plugins can register their component into the home plugin. I think it should be a little structured though. I am not positive if we would want rendering anything by these plugins.

Also: @kgcreative, @BSFishy, we still need a new issue and overall proposal for the home plugin updates correct? Or is that available somewhere?

**Comment by raintygao** [comment]
> That touches upon this. Where 3rd party plugins can register their component into the home plugin. I think it should be a little structured though. I am not positive if we would want rendering anything by these plugins.

@kavilla  Thanks for your reply. I agree that some structure is needed, I prefer to let the plugin to render because component may need to bind some functions from plugin and depend on the context of plugin, otherwise we may need a service that supports plugin to pass components with the handler from plugin(or be able to interact with plugin) to the home.

> I agree! I created a sub-issue: #5589 for the new home plugin view.

I took a look at this code link, it seems to use `service.chrome` to pass pictures, do you think we can follow this to implement here? I'm a little concerned about whether components or some functions or data containing context should be passed through a serialization service. Do you have any ideas?

**Comment by raintygao** [comment]
> I agree! I created a sub-issue: #5589 for the new home plugin view.
> 
> That touches upon this. Where 3rd party plugins can register their component into the home plugin. I think it should be a little structured though. I am not positive if we would want rendering anything by these plugins.
> 
> Also: @kgcreative, @BSFishy, we still need a new issue and overall proposal for the home plugin updates correct? Or is that available somewhere?
> 
> cc: @wbeckler

Also, I would like to know if core team has any plans, I'm glad to implement this, but if the core team is already working on it, please let me know if there is anything I can help. This feature may need to be available in version 2.13.

**Comment by raintygao** [comment]
> I agree! I created a sub-issue: #5589 for the new home plugin view.
> 
> That touches upon this. Where 3rd party plugins can register their component into the home plugin. I think it should be a little structured though. I am not positive if we would want rendering anything by these plugins.
> 
> Also: @kgcreative, @BSFishy, we still need a new issue and overall proposal for the home plugin updates correct? Or is that available somewhere?
> 
> cc: @wbeckler

Also, I would like to know if core team has any plans on this, I'm glad to implement this, but if the core team is already working on it, please let me know if there is anything I can help. This feature may need to be available in version 2.13. Thanks.

**Comment by raintygao** [comment]
Hi @BSFishy , i read this RFC https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5251 and came up with some questions:
1. What's the progress of the implementation of this RFC? Can we have a try on it now?
2. Can this RFC match the requirements in this issue?  If the plugin passes the inputSearchBar component in `categories`,  how can we render it in right specific position of homepage? Such as the position of the rectangle in the screenshot.
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/59455aab-2f43-4487-9ed8-2074165a3e12)
![image](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/42465692/6541e596-c426-4fe2-aa00-c47d39bfb3c7)

**Comment by kgcreative** [comment]
@BSFishy + @kavilla -- i think we need to do a fast follow on the updated homepage design for 2.12, with the idea that we will incrementally add configurability and additional functionality

**Comment by raintygao** [comment]
@kavilla @BSFishy Can anyone have any updates?
